rpc/gnmi/gnmi-path-strings.md

```diff
@@ -0,0 +1,213 @@
+## Representing gNMI Paths as Strings
+
+**Authors**: 
+robjs<sup>⸸</sup>, aashaikh<sup>⸸</sup>, csl<sup>⸸</sup>, hines<sup>⸸</sup>.  
+<sup>⸸</sup> @google.com
+
+October 2018
+
+
+## Overview
+
+This document provides a specification for the representation of a gNMI
+[`Path`](https://github.com/openconfig/gnmi/blob/master/proto/gnmi/gnmi.proto#L133)
+message as a string. String encoding MUST NOT be used within the protocol, but
+provides means to simplify human interaction with gNMI paths - for example, in
+device or daemon configuration.
+
+
+## String Representation of a Single Element
+
+A gNMI path consists of a series of `PathElem` messages.  These messages
+consist of a `name` and a map of `keys`. Each `PathElem` is represented as a
+string in the form:
+
+```
+name[key1=val1][key2=val2]
+```
+
+Where `name` is the `PathElem` `name` field, `key1` represents a key of the
+`keys` map, and `val1` represents the value corresponding to `key1` within the
+map. Where multiple keys are present (as shown with `key1` and `key2`) the
+values are encoded in the form:
+
+```
+[key=value]
+```
+
+and appended to the string. If the key value contains a `]` character, it is
+escaped by preceding it with the backslash (`\`) character. Only `]` and `\`
+characters must be escaped. Newlines and carriage returns are encoded as `\n`
```

Comment:
Is `\` really escaped and unescaped in the reference implementation? It seems like only `\` followed by `]` is handled, and all other occurrences of `\` are left untouched. Which one is right?